wordpress-mobile/gutenberg-mobile
# Issue/874 fix nested lists handling

Fixes: #874

Related GB PR: https://github.com/WordPress/gutenberg/pull/15566
Relate Aztec PR: https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1181

This PR updates the list code to support the following:
 - Activates the indent/outdent buttons on the UI for the list block
 - Improves the handling of the Enter character and make sure the format of text between Aztec and the format library is consistent 
 - Makes the handling of delete of new lines to be handled by the rich text format structure, instead of the native side.

How to test:
 - First of all try, the issue stated on #874 to see if it's working correctly
 - Then try to use the indent and outdent buttons and see if they work correctly
 - Try to delete list items and see if it works correctly
